x-pack/plugins/security_solution/server/endpoint/services/actions/clients/crowdstrike/crowdstrike_actions_client.ts

```diff
@@ -0,0 +1,341 @@
+  private readonly connectorActionsClient: NormalizedExternalConnectorClient;
+
+  constructor({ connectorActions, ...options }: CrowdstrikeActionsClientOptions) {
+    super(options);
+    this.connectorActionsClient = connectorActions;
+    connectorActions.setup(CROWDSTRIKE_CONNECTOR_ID);
+  }
+
+  protected async writeActionRequestToEndpointIndex<
+    TParameters extends EndpointActionDataParameterTypes = EndpointActionDataParameterTypes,
+    TOutputContent extends EndpointActionResponseDataOutput = EndpointActionResponseDataOutput,
+    TMeta extends {} = {}
+  >(
+    actionRequest: ResponseActionsClientWriteActionRequestToEndpointIndexOptions<
+      TParameters,
+      TOutputContent,
+      TMeta
+    >
+  ): Promise<
+    LogsEndpointAction<TParameters, TOutputContent, TMeta & CrowdstrikeActionRequestCommonMeta>
+  > {
+    const agentId = actionRequest.endpoint_ids[0];
+    const eventDetails = await this.getEventDetailsById(agentId);
+
+    const hostname = eventDetails.crowdstrike.event.HostName;
+    return super.writeActionRequestToEndpointIndex({
+      ...actionRequest,
+      hosts: {
+        [agentId]: { name: hostname },
+      },
+      meta: {
+        hostName: hostname,
+        ...(actionRequest.meta ?? {}),
+      } as TMeta & CrowdstrikeActionRequestCommonMeta,
+    });
+  }
+
+  /**
+   * Sends actions to Crowdstrike directly (via Connector)
+   * @private
+   */
+  private async sendAction(
+    actionType: SUB_ACTION,
+    actionParams: object
+  ): Promise<ActionTypeExecutorResult<unknown>> {
+    const executeOptions: NormalizedExternalConnectorClientExecuteOptions = {
+      params: {
+        subAction: actionType,
+        subActionParams: actionParams,
+      },
+    };
+
+    this.log.debug(
+      `calling connector actions 'execute()' for Crowdstrike with:\n${stringify(executeOptions)}`
+    );
+
+    const actionSendResponse = await this.connectorActionsClient.execute(executeOptions);
+
+    if (actionSendResponse.status === 'error') {
+      this.log.error(stringify(actionSendResponse));
+
+      throw new ResponseActionsClientError(
+        `Attempt to send [${actionType}] to Crowdstrike failed: ${
+          actionSendResponse.serviceMessage || actionSendResponse.message
+        }`,
+        500,
+        actionSendResponse
+      );
+    } else {
+      this.log.debug(`Response:\n${stringify(actionSendResponse)}`);
+    }
+
+    return actionSendResponse;
+  }
+
+  private async getEventDetailsById(agentId: string): Promise<{
+    crowdstrike: { event: { HostName: string } };
+  }> {
+    const search = {
+      index: ['logs-crowdstrike.fdr*', 'logs-crowdstrike.falcon*'],
+      fields: [{ field: '*' }],
+      _source: false,
+      body: {
+        query: {
+          bool: {
+            filter: [{ term: { 'crowdstrike.event.DeviceId': agentId } }],
+          },
+        },
+      },
+    };
+    const result = await this.options.esClient
```

Comment:
Yeah, I saw that. I guess thats ok, but it would be best to add it to the variable here for clarity and in order to understand the code better. I'll let you decide which way to go.

Reply:
set type at the const level 👍

Reply:
reverted... casting it as {crowdstrike} bring less trouble, because the expandDottedObject may return {} by default...